backend/src/main/java/com/redhat/cloud/notifications/auth/kessel/KesselAssets.java

```diff
@@ -0,0 +1,154 @@
+
+        // Measure the time it takes to perform the operation with Kessel.
+        final Timer.Sample createIntegrationTimer = Timer.start(this.meterRegistry);
+
+        // Send the request to the inventory.
+        final CreateNotificationsIntegrationResponse response;
+        try {
+            response = this.notificationsIntegrationClient.CreateNotificationsIntegration(request);
+        } catch (final StatusRuntimeException e) {
+            Log.errorf(
+                e,
+                "[identity: %s][workspace_id: %s][integration_id: %s] Unable to create integration in Kessel's inventory",
+                SecurityContextUtil.extractRhIdentity(securityContext), workspaceId, integrationId, request
+            );
+
+            throw e;
+        }
+
+        // Stop the timer.
+        createIntegrationTimer.stop(this.meterRegistry.timer(KESSEL_METRICS_INVENTORY_INTEGRATION_TIMER_NAME, Tags.of(Constants.KESSEL_METRICS_TAG_RESOURCE_TYPE_KEY, ResourceType.INTEGRATION.getKesselName())));
+
+        Log.tracef("[identity: %s][workspace_id: %s][integration_id: %s] Received payload for the integration creation in Kessel's inventory: %s", SecurityContextUtil.extractRhIdentity(securityContext), workspaceId, integrationId, response);
+        Log.debugf("[identity: %s][workspace_id: %s][integration_id: %s] Integration created in Kessel's inventory", SecurityContextUtil.extractRhIdentity(securityContext), workspaceId, integrationId);
+    }
+
+    /**
+     * Deletes an integration from Kessel's inventory.
+     * @param securityContext the security context which holds the subject of
+     *                        the operation.
+     * @param workspaceId the identifier for the workspace the integration will
+     *                    be deleted from.
+     * @param integrationId the identifier of the integration we deleted on
+     *                      our side.
+     */
+    public void deleteIntegration(final SecurityContext securityContext, final String workspaceId, final String integrationId) {
+        // Build the request for Kessel's inventory.
+        final DeleteNotificationsIntegrationRequest request = this.buildDeleteIntegrationRequest(integrationId);
+
+        Log.tracef("[identity: %s][workspace_id: %s][integration_id: %s] Payload for deleting the integration in Kessel's inventory: %s", SecurityContextUtil.extractRhIdentity(securityContext), workspaceId, integrationId, request);
+
+        // Measure the time it takes to perform the operation with Kessel.
+        final Timer.Sample deleteIntegrationTimer = Timer.start(this.meterRegistry);
+
+        // Send the request to the inventory.
+        final DeleteNotificationsIntegrationResponse response;
+        try {
+            response = this.notificationsIntegrationClient.DeleteNotificationsIntegration(request);
+        } catch (final StatusRuntimeException e) {
```

Comment:
We could catch `Exception` here.

Reply:
I suppose `StatusRuntimeException` is used to let us know that something went wrong in Kessel while processing our request. Is that correct?

Reply:
That is correct, yes. I think we can modify that and set the generic `Exception` catch because exception mapper will catch the particular `StatusRuntimeException` and handle it properly anyway.